lib/engine/route.rb

```diff
@@ -152,8 +152,8 @@ def check_cycles!
     end
 
     def check_overlap!
-      @routes.flat_map(&:paths).group_by(&:itself).each do |k, v|
-        @game.game_error("Route cannot use same path twice #{k.inspect}") if v.size > 1
+      @routes.flat_map { |r| r&.paths&.flat_map(&:track_resources_used) }.group_by(&:itself).each do |k, v|
```

Comment:
how can route be nil? i think we should switch it to 'bare metal', it's simple enough

@routes.flat_map do |route|
route.paths.flat_map ...